### ISSUE TITLE
run-vcpkg@10 - Backslash in path under Windows are stripped.

### ISSUE THREAD

**Issue by bansan85** [lukka/run-vcpkg#144]
When migrating from v6 to v10, I can't use anymore `${{ github.workspace }}` under Windows for parameter `vcpkgJsonGlob` (for example)

For example I needed to hardcode path:

```
--- a/.github/workflows/build-windows.yml
+++ b/.github/workflows/build-windows.yml
@@ -15,6 +15,10 @@ jobs:
             cc: cl
             cxx: cl
             os: windows-latest
+    env:
+      VCPKG_DEFAULT_TRIPLET: 'x64-windows'
+      # Need to hardcode path to avoid escape delimiter path
+      VCPKG_INSTALLED_DIR: 'D:/a/jessica/jessica/vcpkg/installed'
     steps:
       - name: Stop if format failed
         if: ${{ github.event.workflow_run.conclusion != 'success' }}
@@ -35,13 +39,10 @@ jobs:
           key: ${{ secrets.GITHUBCI_PRIVATE_KEY }}
           known_hosts: github.com ssh-rsa ${{ secrets.GITHUBCI_KNOW_HOSTS_WINDOWS }}
       - name: Install Windows Vcpkg
-        uses: lukka/run-vcpkg@v6
-        id: runvcpkg
+        uses: lukka/run-vcpkg@v10
         with:
-          vcpkgArguments: '@${{ github.workspace }}/config/vcpkg_windows.txt'
-          vcpkgDirectory: '${{ github.workspace }}/vcpkg'
-          vcpkgTriplet: 'x64-windows'
-          appendedCacheKey: ${{ hashFiles( format('{0}/config/vcpkg_windows.txt', github.workspace) ) }}
+          vcpkgJsonGlob: 'D:/a/jessica/jessica/config/windows/vcpkg.json'
+          runVcpkgInstall: true
       - name: CMake
         run: |
           mkdir "${{ github.workspace }}/build"
```

If I use `${{ github.workspace }}`, vcpkg will fail with message:

```
Error: run-vcpkg action execution failed: Cannot run 'install,--recurse,--clean-after-build,--x-install-root,D:ajessicajessica/vcpkg/installed,--triplet,x64-windows' because no valid directory has been provided.
```

As you can see, all backslashes has been stripped.

**Comment by lukka** [comment]
@bansan85 yes, I can see that in this run of the [tests that run a similar scenario](https://github.com/lukka/run-vcpkg/runs/7268682176?check_suite_focus=true#step:4:241). Not sure yet what the fix could be looking like, suggestions and ideas are welcome.

**Comment by lukka** [comment]
@bansan85 would you be able to try your workflow with `run-vcpkg@dev/issue144` ?

**Comment by lukka** [comment]
@bansan85  thanks for trying this out! The error now is related to the action not being able to find the vcpkg.json file, hence it does not know where to run `vcpkg.exe` from, i.e.,
![image](https://user-images.githubusercontent.com/681992/178628761-45a34ed2-226f-43c5-9f30-2790db240311.png)
Since i can clearly the `vcpkg.json` file is in your repo at that path, clearly my fix needs "improvements".

**Comment by lukka** [comment]
@bansan85 thanks again for reporting this issue! 
Regarding the runs you provided at https://github.com/lukka/run-vcpkg/issues/144#issuecomment-1182428101, there is nothing to do on the `run-vcpkg` side, since the problem is that the `${{ github.workspace }}` contains backslashes on Windows. When that variable is used to provide the value for a parameter which is an arbitrary string (e.g., [`vcpkgJsonGlob` parameter](https://github.com/lukka/run-vcpkg/blob/dev/issue144/action.yml#L39)) the action must not manipulate the inpu in any way before using it, but it must use it as the user provided it.

The solution for you is to use instead `**/config/windows/vcpkg.json`.

I added the documentation about this on the [action.yml](https://github.com/lukka/run-vcpkg/blob/dev/issue144/action.yml#L39).

Let me know if this works for you, and if it makes sense as well, thanks.

**Comment by bansan85** [comment]
I'm out for the week. I will keep you in touch.

**Comment by bansan85** [comment]
It looks to work 😃 : https://github.com/bansan85/jessica/runs/7386894291

I can use `VCPKG_INSTALLED_DIR: '${{ github.workspace }}\vcpkg\installed'` but I need to use `vcpkgJsonGlob: '**/config/windows/vcpkg.json'`.

It's sad that I could use `vcpkgArguments: '@${{ github.workspace }}/config/vcpkg_windows.txt'`or `vcpkgDirectory: '${{ github.workspace }}/vcpkg'` in `v6` but not in `v10`.

**Comment by lukka** [comment]
Thanks for confirming it!

Fixed in release `v10.3`: https://github.com/lukka/run-vcpkg/releases/tag/v10.3
